### ISSUE TITLE
Context has no highlighting when first opening Neovim

### ISSUE THREAD

**Issue by MariaSolOs** [nvim-treesitter/nvim-treesitter-context#387]
### Description

If I open Neovim in a way that the context window is already open, then the context content has no highlighting until the context window is closed at least once.

### Neovim version

v0.10.0-dev-2257+g302fa5f7d

### Expected behavior

For the context window to always have the expected highlighting.

### Actual behavior

This is what it looks like the first time:
![20240205_13h26m03s_grim](https://github.com/nvim-treesitter/nvim-treesitter-context/assets/62502207/9bdaaa12-352e-4429-aade-0978015f9ee2)

### Minimal config

```Lua
local plugins = {
  ts = "https://github.com/nvim-treesitter/nvim-treesitter",
  ts_context = "https://github.com/nvim-treesitter/nvim-treesitter-context",
  tokyo_night = "https://github.com/folke/tokyonight.nvim",
}

for name, url in pairs(plugins) do
  local install_path = "nvim_issue/" .. name
  if vim.fn.isdirectory(install_path) == 0 then
    vim.fn.system { "git", "clone", "--depth=1", url, install_path }
  end
  vim.o.runtimepath = install_path .. "," .. vim.o.runtimepath
end

vim.cmd [[colorscheme tokyonight]]

require("tokyonight").setup()
require("nvim-treesitter.configs").setup { ensure_installed = { "lua" } }
require("treesitter-context").setup()

local foo = {
  bar = {
    ensure_installed = {
      "bash",
      "c",
      "cpp",
      "fish",
      "gitcommit",
      "javascript",
      "json",
      "json5",
      "jsonc",
      "lua",
      "markdown",
      "markdown_inline",
      "python",
      "query",
      "regex",
      "rust",
      "scss",
      "toml",
      "typescript",
      "vim",
      "vimdoc",
      "bash",
      "c",
      "cpp",
      "fish",
      "gitcommit",
      "javascript",
      "json",
      "json5",
      "jsonc",
      "lua",
      "markdown",
      "markdown_inline",
      "python",
      "query",
      "regex",
      "rust",
      "scss",
      "toml",
      "typescript",
      "vim",
      "vimdoc",
    },
  },
}
```

### Steps to reproduce

1. Include the above configuration file in `minimal.lua`.
2. Open Neovim with `nvim --clean -u minimal.lua minimal.lua -c "normal50Gzz"`.
3. Notice how the context window has no colorization. Then run `gg50Gzz` and notice how the context window is now highlighted.

**Comment by Danielkonge** [comment]
I have run into a similar issue in longer python files when I scroll up into a method in a longer class. I think the cause is the upstream tree-sitter updates, which saves some work for off screen text in certain cases (based on my understanding). 

Your problem is probably manageable by this plugin, but I think it will be hard to avoid some problems with the current tree-sitter code upstream. (Though I might misunderstand something here. This problem doesn't annoy me too much usually, so I haven't really looked into it.)

**Comment by lewis6991** [comment]
> I have run into a similar issue in longer python files when I scroll up into a method in a longer class. I think the cause is the upstream tree-sitter updates, which saves some work for off screen text in certain cases (based on my understanding).
>
> Your problem is probably manageable by this plugin, but I think it will be hard to avoid some problems with the current tree-sitter code upstream. (Though I might misunderstand something here. This problem doesn't annoy me too much usually, so I haven't really looked into it.)

Unlikely. These updates only apply to injections.

I think the issue is that nvim-treesitter-context is running before the `TreesitterHighlighter` has been setup. Not sure for certain though.

I've raised #392 which simply avoids using `vim.treesitter.highlighter` and instead we use our own queries. Let me know if that resolves things.

**Comment by romgrk** [comment]
I had to revert the fix due to #410, but I was unable to reproduce the issue with the minimal config above.

@MariaSolOs Can you confirm if you're still seeing the issue with the latest commit?

**Comment by MariaSolOs** [comment]
@romgrk Yes, I can still reproduce it.
Here's a video:


https://github.com/nvim-treesitter/nvim-treesitter-context/assets/62502207/5b9f253b-633e-4b8e-ab40-5a3bb8dbee73
